acceptance/api/v1/service_delete_test.go

```diff
@@ -0,0 +1,154 @@
+package v1_test
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/epinio/epinio/acceptance/helpers/catalog"
+	"github.com/epinio/epinio/acceptance/helpers/proc"
+	v1 "github.com/epinio/epinio/internal/api/v1"
+	"github.com/epinio/epinio/internal/services"
+	"github.com/epinio/epinio/pkg/api/core/v1/models"
+	helmapiv1 "github.com/k3s-io/helm-controller/pkg/apis/helm.cattle.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ServiceDelete Endpoint", func() {
+	var namespace string
+	var catalogService models.CatalogService
+
+	When("namespace doesn't exist", func() {
+		It("returns 404", func() {
+			endpoint := fmt.Sprintf("%s%s/namespaces/notexists/services/whatever", serverURL, v1.Root)
+			response, err := env.Curl("DELETE", endpoint, strings.NewReader(""))
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(response.StatusCode).To(Equal(http.StatusNotFound))
+		})
+	})
+
+	When("namespace exists", func() {
+		BeforeEach(func() {
+			namespace = catalog.NewNamespaceName()
+			env.SetupAndTargetNamespace(namespace)
+
+			catalogService = models.CatalogService{
+				Name:      catalog.NewCatalogServiceName(),
+				HelmChart: "nginx",
+				HelmRepo: models.HelmRepo{
+					Name: "",
+					URL:  "https://charts.bitnami.com/bitnami",
+				},
+				Values: "{'service': {'type': 'ClusterIP'}}",
+			}
+			createCatalogService(catalogService)
+		})
+
+		AfterEach(func() {
+			deleteCatalogService(catalogService.Name)
+			env.DeleteNamespace(namespace)
+		})
+
+		When("service instance doesn't exist", func() {
+			It("returns 404", func() {
+				endpoint := fmt.Sprintf("%s%s/namespaces/%s/services/notexists", serverURL, v1.Root, namespace)
+				response, err := env.Curl("DELETE", endpoint, strings.NewReader(""))
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(response.StatusCode).To(Equal(http.StatusNotFound))
+			})
+		})
+
+		When("helmchart exists", func() {
+			var serviceName string
+
+			BeforeEach(func() {
+				serviceName = catalog.NewServiceName()
+			})
+
+			When("helmchart is not labeled", func() {
+				BeforeEach(func() {
+					helmChart := helmapiv1.HelmChart{
```

Comment:
Should we use the MakeInstance func that you already added with the other PR?

Reply:
I need to control the labels in this case (see the `when` description)

Reply:
I can use the helper in the "labeled" case below though. Changing it.